packages/atomic/src/components/search/atomic-search-interface/store.ts

```diff
@@ -0,0 +1,116 @@
+import {
+  NumericFacetValue,
+  DateFacetValue,
+  SortCriterion,
+} from '@coveo/headless';
+import {VNode} from '@stencil/core';
+import {makeDesktopQuery} from '../atomic-layout/search-layout';
+import {DEFAULT_MOBILE_BREAKPOINT} from '../../../utils/replace-breakpoint';
+import {
+  createAtomicCommonStore,
+  AtomicCommonStoreData,
+} from '../../common/search-interface/store';
+
+interface FacetInfo {
+  label: string;
+}
+
+export interface ResultListInfo {
+  focusOnNextNewResult(): void;
+}
+
+interface FacetValueFormat<ValueType> {
+  format(facetValue: ValueType): string;
+  content?(facetValue: ValueType): VNode;
+}
+
+type FacetType = 'facets' | 'numericFacets' | 'dateFacets' | 'categoryFacets';
+type FacetStore<F extends FacetInfo> = Record<string, F>;
+
+export interface SortDropdownOption {
+  expression: string;
+  criteria: SortCriterion[];
+  label: string;
+}
+
+export interface AtomicStoreData extends AtomicCommonStoreData {
+  facets: FacetStore<FacetInfo>;
+  numericFacets: FacetStore<FacetInfo & FacetValueFormat<NumericFacetValue>>;
+  dateFacets: FacetStore<FacetInfo & FacetValueFormat<DateFacetValue>>;
+  categoryFacets: FacetStore<FacetInfo>;
+  facetElements: HTMLElement[];
+  sortOptions: SortDropdownOption[];
+  mobileBreakpoint: string;
+  fieldsToInclude: string[];
+  resultList?: ResultListInfo;
+}
+
+export function createAtomicStore() {
+  const commonStore = createAtomicCommonStore<AtomicStoreData>({
+    loadingFlags: [],
+    facets: {},
+    numericFacets: {},
+    dateFacets: {},
+    categoryFacets: {},
+    facetElements: [],
+    sortOptions: [],
+    iconAssetsPath: '',
+    mobileBreakpoint: DEFAULT_MOBILE_BREAKPOINT,
+    fieldsToInclude: [],
+  });
+
+  return {
+    ...commonStore,
+    registerFacet<T extends FacetType, U extends string>(
+      facetType: T,
+      data: AtomicStoreData[T][U] & {facetId: U; element: HTMLElement}
+    ) {
+      if (commonStore.state[facetType][data.facetId]) {
+        return;
+      }
+
+      commonStore.state[facetType][data.facetId] = data;
+      commonStore.state.facetElements.push(data.element);
+    },
+
+    registerResultList(data: ResultListInfo) {
+      commonStore.set('resultList', data);
+    },
+
+    getFacetElements() {
+      return commonStore.state.facetElements.filter((element) =>
+        isInDocument(element)
+      );
+    },
+
+    getAllFacets() {
+      return {
+        ...commonStore.state.facets,
+        ...commonStore.state.dateFacets,
+        ...commonStore.state.categoryFacets,
+        ...commonStore.state.numericFacets,
+      };
+    },
+
+    isMobile() {
```

Comment:
I prefer to leave it here since, so far, we (obviously) did not need it for other specific interface. When it's needed, we can move it. It's not complicated.

My logic is that I want to leave common store as barebone as possible, and not tackle things that we might or might not need later. Because it's easier to "move down" a function than it is to remove it.